PROPOSAL.md

```diff
@@ -0,0 +1,100 @@
+# Proposal for Refactoring Backtesting
+
+Reference: #DARTS-133
+
+---
+
+## Summary
+
+The main idea behind this proposal is to refactor `backtesting.py` by implementing it directly in the model 
+class. Here would be a quick list of changes proposed:
+
+- `backtest_forecasting`, `backtest_regression`, `backtest_gridsearch` -> moved in corresping model methods
```

Comment:
Do you think `backtest_gridsearch` could be made to a static method of `ForecastingModel`? (Since it takes a class, and not an instance as argument)

Reply:
How about having an abstract method `ForecastingModel.hyperparams()` that returns the params dict that `backtest_gridsearch()` would iterate over? This way model implementers would just have to implement this one method for the model to be grid-searchable. @guillaumeraille @pennfranc wdyt?

Reply:
Not a bad idea but I think the params dict is more dataset / application dependent hence unkown at model implementation time dont you think ?

Reply:
Right that's true. What I had in mind was more for this method to return decent default hyper-params, but we would still need to accept user-defined parameters.